cpp/src/cider/exec/nextgen/operators/VectorizedProjectNode.cpp

```diff
@@ -181,68 +284,171 @@ jitlib::JITValuePointer& allocateNullBuffer(context::CodegenContext& context,
   return nullable_values.getNull();
 }
 
+void generateNonBoolInputExprsGroupCode(context::CodegenContext& context,
+                                        VectorizedProjectTranslator::ExprsGroup& group) {
+  // Utilize C2R and R2C to generate column load and store.
+  auto c2r_node = createOpNode<ColumnToRowNode>(group.input_exprs, true);
+  auto c2r_translator = c2r_node->toTranslator();
+  auto r2c_node = createOpNode<RowToColumnNode>(group.exprs, c2r_node.get());
+  auto r2c_translator = r2c_node->toTranslator();
+
+  {
+    // Temporarily set nullable of all exprs in output expr tree as false.
+    NullableGuard null_guard(group.exprs);
+
+    c2r_translator->codegen(context,
+                            [&group, &r2c_translator](context::CodegenContext& context) {
+                              // Generate Project code.
+                              for (auto& expr : group.exprs) {
+                                expr->codegen(context);
+                              }
+                              r2c_translator->consume(context);
+                            });
+  }
+
+  // Process null vector.
+  // TBD   (bigPYJ1151): Currently, only support trival null processing (e.g. merge all
+  // leaf exprs' null vector by bitwise-and).
+  // TODO (bigPYJ1151): Specialize more null vector 'and' primary function to reduce
+  // function-call overhead and load/store inst.
```

Comment:
Can we collect all input columns' null as vector, and remove which is null, then BitAnd all other columns?

Reply:
Good idea.

Reply:
then we can remove `mutableRawNulls` in `CiderPipelineOperator::addInput`

Reply:
> then we can remove `mutableRawNulls` in `CiderPipelineOperator::addInput`

For the last row-based data-centric loop, null vectors are also needed.

Reply:
check it dynamically or create a temporary null for the last row-based?

Reply:
Infeasible, because the last row-based data-centric loop fused multiple inputs, so we have to create a check branch for each row input, which may lead to performance issues.